src/tools/rbd_mirror/InstanceWatcher.cc

```diff
@@ -785,8 +785,14 @@ void InstanceWatcher<I>::handle_image_acquire(
   const std::string &peer_image_id, Context *on_finish) {
   dout(20) << "global_image_id=" << global_image_id << dendl;
 
-  m_instance_replayer->acquire_image(global_image_id, peer_mirror_uuid,
-                                     peer_image_id, on_finish);
+  auto ctx = new FunctionContext(
```

Comment:
Probably would be a good idea to track these queued callbacks via an ```AsyncOpTracker``` to prevent lock release / shut down races.